ui_framework/doc_site/src/components/guide_demo/guide_demo.jsx

```diff
@@ -0,0 +1,64 @@
+import React, {
+  Component,
+  PropTypes,
+} from 'react';
+
+import classNames from 'classnames';
+
+export class GuideDemo extends Component {
+  componentDidMount() {
+    this.update();
+  }
+
+  shouldComponentUpdate(nextProps) {
+    return nextProps !== this.props;
```

Comment:
Have you double-check that you don't get a new ref every time here? Should probably just [use `React.PureComponent` instead](https://facebook.github.io/react/docs/react-api.html#react.purecomponent), which does a shallow compare on props and state.